### PR TITLE
Added tab navigation for community feed

### PR DESCRIPTION
**Pull request by lucyxubroad**
## Overview

- Added the top navigation feed to allow for our community feed page

## Changes Made

- Changed our home view controller to be a page controller containing two pages
- Removed the profile button from our match screens and move into our main home page

## Test Coverage

- Manual testing

## Next Steps (delete if not applicable)

- Next steps: Add the community screen

## Related PRs or Issues (delete if not applicable)

#61 

## Screenshots (delete if not applicable)

<details>

  <summary>Screen Shot Name</summary>

  
  

ezgif com-video-to-gif-20

</details>

### CODE REVIEW

File: CoffeeChat/Controller/HomeViewController.swift [hunk 1]
```diff
@@ -82,114 +40,99 @@ class HomeViewController: UIViewController {
...
+        tabLayout.minimumInteritemSpacing = 0
+
+        tabCollectionView = UICollectionView(frame: .zero, collectionViewLayout: tabLayout)
+        tabCollectionView.delegate = self
+        tabCollectionView.dataSource = self
+        tabCollectionView.register(HomeTabOptionCollectionViewCell.self, forCellWithReuseIdentifier: tabCellReuseIdentifier)
+        tabCollectionView.backgroundColor = .white
+        tabCollectionView.clipsToBounds = true
+        tabCollectionView.layer.masksToBounds = false
+        tabCollectionView.layer.cornerRadius = 24
```
Comment (pnor) [inline CoffeeChat/Controller/HomeViewController.swift:58]: The rounding of the sides look a little bit off to me, although the Figma file says 24.
image

I tried it with cornerRadius=20 and got this
image

File: CoffeeChat/Controller/HomeViewController.swift [hunk 2]
```diff
@@ -82,114 +40,99 @@ class HomeViewController: UIViewController {
...
+        // TODO: Fix tab bar shadows
```
Comment (pnor) [inline CoffeeChat/Controller/HomeViewController.swift:59]: address this PR before we merge in or delete if not relevent

File: CoffeeChat/Controller/TabPageViewController.swift [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+//  Created by Lucy Xu on 10/1/20.
+//  Copyright © 2020 cuappdev. All rights reserved.
+//
+
+import UIKit
+
+class TabPageViewController: UIPageViewController {
+
+    // MARK: - Private View Vars
+    private var matchViewController: MatchViewController!
```
Comment (pnor) [inline CoffeeChat/Controller/TabPageViewController.swift:14]: We could avoid force unwrapping these by initializing them in the init, so if we forget to set one it'll be caught at build time

Although we do a similar pattern in other places in the project, so its up to you if you think its worth changing

**Review by pnor — APPROVED** [review]
Looks good, I'll have to resolve merge conflicts on my end though lol

File: CoffeeChat/Controller/HomeViewController.swift [hunk 4]
```diff
@@ -2,76 +2,34 @@
...
-        MatchSummary(title: "He is also part of...", detail: "EzraBox")
-    ]
+    private var tabCollectionView: UICollectionView!
+    private var tabContainerView: UIView!
+    private var tabPageViewController: TabPageViewController!
+    let profileButtonSize = CGSize(width: 35, height: 35)
+
+    // MARK: - Private Data Vars
+    private var activeTabIndex = 0
+    private let tabCellReuseIdentifier = "tabCellReuseIdentifier"
```
Comment (ywsang) [inline CoffeeChat/Controller/HomeViewController.swift:23]: can we move to the cell class file, as a static constant?

File: CoffeeChat/Cells/HomeTabOptionCollectionViewCell.swift [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+//  Copyright © 2020 cuappdev. All rights reserved.
+//
+
+import UIKit
+
+class HomeTabOptionCollectionViewCell: UICollectionViewCell {
+
+    private let tabLabel = UILabel()
+    private var activeCellColor: UIColor = .darkGreen
+    private var inactiveCellColor: UIColor = .inactiveGreen
```
Comment (ywsang) [inline CoffeeChat/Cells/HomeTabOptionCollectionViewCell.swift:15]: nitpick but can we the var names to be `...TextColor`? just a little more accurate description wise

**Review by ywsang — APPROVED** [review]
lgtm! just a little thing about the reuse ids, though i could be tripping about that

File: CoffeeChat/Controller/MatchViewController.swift [hunk 6]
```diff
@@ -0,0 +1,137 @@
...
+
+    private let matchDemographicsLabel = UILabel()
+    private let matchNameLabel = UILabel()
+    private let matchProfileImageView = UIImageView()
+    private let matchSummaryTableView = UITableView()
+    private let reachOutButton = UIButton()
+
+    private let imageSize = CGSize(width: 120, height: 120)
+    private let reachOutButtonSize = CGSize(width: 200, height: 50)
+    private let cellReuseId = "cellReuseIdentifier"
```
[the remark this answers is not in the document]
Comment (lucyxubroad) [reply CoffeeChat/Controller/MatchViewController.swift:22]: Oh LOL, we do.... but uh that's maybe a for later refactor... ill make a PR for that but will fix the reuse ID

File: CoffeeChat/Controller/HomeViewController.swift [hunk 7]
```diff
@@ -82,114 +40,98 @@ class HomeViewController: UIViewController {
...
 
-    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
-        guard let cell = tableView.dequeueReusableCell(withIdentifier: cellReuseId, for: indexPath) as? MatchSummaryTableViewCell else { return UITableViewCell() }
-        let summary = matchSummaries[indexPath.row]
-        cell.configure(for: summary)
+    func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
+        guard let cell = collectionView.dequeueReusableCell(withReuseIdentifier: tabCellReuseIdentifier, for: indexPath) as? HomeTabOptionCollectionViewCell else { return UICollectionViewCell() }
+        if indexPath.item == activeTabIndex {
+            cell.isSelected = true
+        }
```
Comment (camruss) [inline CoffeeChat/Controller/HomeViewController.swift:126]: `cell.isSelected = indexPath.item == activeTabIndex` unless you want it to stay true even after `indexPath.item == activeTabIndex` is no longer true.

**Review by camruss — APPROVED** [review]
looks good (question for the designers tho: should there be a little space at the top below the UISegmentControl when changing screens bc. it's pretty abrupt imo)

File: CoffeeChat/Controller/HomeViewController.swift (hunk 7 above)
Comment (lucyxubroad) [reply CoffeeChat/Controller/HomeViewController.swift:125]: Oh ya so, I its just cause I made the other screen a diff color to show the page scroll. The actual page will have the same color and will have more space at the top so it shouldn't look so abrupt.

**Pull request merged**